lib/Demangling/Demangler.cpp

```diff
@@ -113,6 +113,31 @@ namespace swift {
 namespace Demangle {
 
 //////////////////////////////////
+// Public utility functions    //
+//////////////////////////////////
+
+int getManglingPrefixLength(const char *mangledName) {
```

Comment:
It's generally better to use qualified declarations when implementing something declared in a header, because you'll actually get signature-match checking instead of accidentally declaring an overload.  That is, instead of defining this as:

```
  namespace Demangle {
    int Demangle::getManglingPrefixLength(const char *mangledName) {
      ...
    }
  }
```

it should be declared as::

```
  int Demangle::getManglingPrefixLength(const char *mangledName) {
    ...
  }
```